src/parameterizations/lateral/MOM_hor_visc.F90

```diff
@@ -531,20 +501,34 @@ subroutine horizontal_viscosity(u, v, h, diffu, diffv, MEKE, VarMix, G, GV, US,
 
     I_GME_h0 = 1.0 / CS%GME_h0
     do j=Jsq-1,Jeq+2 ; do i=Isq-1,Ieq+2
-      if (grad_vel_mag_bt_h(i,j)>0) then
+      boundary_mask_h = (G%mask2dCu(I,j) * G%mask2dCv(i,J) * G%mask2dCu(I-1,j) * G%mask2dCv(i,J-1))
+      grad_vel_mag_bt_h = G%mask2dT(I,J) * boundary_mask_h * (dudx_bt(i,j)**2 + dvdy_bt(i,j)**2 + &
+            (0.25*((dvdx_bt(I,J)+dvdx_bt(I-1,J-1)) + (dvdx_bt(I,J-1)+dvdx_bt(I-1,J))))**2 + &
+            (0.25*((dudy_bt(I,J)+dudy_bt(I-1,J-1)) + (dudy_bt(I,J-1)+dudy_bt(I-1,J))))**2)
+      ! Probably the following test could be simplified to
+      ! if (boundary_mask_h * G%mask2dT(I,J) > 0.0) then
+      if (grad_vel_mag_bt_h > 0.0) then
         GME_effic_h(i,j) = CS%GME_efficiency * G%mask2dT(I,J) * &
             (MIN(htot(i,j) * I_GME_h0, 1.0)**2)
       else
         GME_effic_h(i,j) = 0.0
       endif
     enddo ; enddo
 
-    do J=Jsq-2,Jeq+1 ; do I=Isq-2,Ieq+1
-      if (grad_vel_mag_bt_q(I,J)>0) then
+    do J=js-2,je+1 ; do I=is-2,ie+1
+      boundary_mask_q = (G%mask2dCv(i,J) * G%mask2dCv(i+1,J) * G%mask2dCu(I,j) * G%mask2dCu(I,j+1))
+      grad_vel_mag_bt_q = G%mask2dBu(I,J) * boundary_mask_q * (dvdx_bt(I,J)**2 + dudy_bt(I,J)**2 + &
+            (0.25*((dudx_bt(i,j)+dudx_bt(i+1,j+1)) + (dudx_bt(i,j+1)+dudx_bt(i+1,j))))**2 + &
+            (0.25*((dvdy_bt(i,j)+dvdy_bt(i+1,j+1)) + (dvdy_bt(i,j+1)+dvdy_bt(i+1,j))))**2)
+      ! Probably the following test could be simplified to
+      ! if (boundary_mask_q * G%mask2dBu(I,J) > 0.0) then
+      if (grad_vel_mag_bt_q > 0.0) then
         h_arith_q = 0.25 * ((htot(i,j) + htot(i+1,j+1)) + (htot(i+1,j) + htot(i,j+1)))
         I_hq = 1.0 / h_arith_q
         h_harm_q = 0.25 * h_arith_q * ((htot(i,j)*I_hq + htot(i+1,j+1)*I_hq) + &
                                        (htot(i+1,j)*I_hq + htot(i,j+1)*I_hq))
+        !### The two expressions above looks like they simplify to just the arithmetic mean!
```

Comment:
Good catch! Indeed, these expressions simplify to the arithmetic mean in line 528. However, answers changed when I used `h_harm_q = 0.25 * ((htot(i,j) + htot(i+1,j+1)) + (htot(i+1,j) + htot(i,j+1)))`. Any idea why? Regardless, we do not need to maintain answers for USE_GME = True` at this time, so I think we should we the simplified expression.

Reply:
The reason why the answers change is simple.  At roundoff, the expression a * (1/a) is not exactly one.  We have seen this many times where replacing division by the multiplication by the reciprocal of the denominator changes answers at roundoff.  However, the guidance that we can simplify this expression now (before it is more widely used) and can tolerate answer changes is perfect.  I will go ahead and make the simplification.